lib/ansible/plugins/lookup/first_found.py

```diff
@@ -32,27 +32,55 @@
       skip:
         type: boolean
         default: False
-        description: Return an empty list if no file is found, instead of an error.
+        description:
+          - When C(True), return an empty list when no files are matched.
+          - This is useful when used with C(with_first_found), as an empty list return to C(with_) calls
+            causes the calling task to be skipped.
+          - When used as a template via C(ookup) or C(query), setting I(skip=True) will *not* cause the task to skip.
+            Tasks must handle the empty list return from the template.
+          - When C(False) and C(lookup) or C(query) specifies  I(errors='ignore') all errors (including no file found,
+            but potentially others) return an empty string or an empty list respectively.
+          - When C(True) and C(lookup) or C(query) specifies I(errors='ignore'), no file found will returni an empty
+            list and other potential errors return an empty string or empty list depending on the template call (i.e. return
+            values of C(lookup) v C(query)).
 """
 
 EXAMPLES = """
-- name: show first existing file or ignore if none do
-  debug:
-    msg: "{{ lookup('first_found', findme, errors='ignore') }}"
+- name: Set _found_file to the first existing file, raising an error if a file is not found
+  set_fact:
+    _found_file: "{{ lookup('first_found', findme) }}"
   vars:
     findme:
       - /path/to/foo.txt
       - bar.txt  # will be looked in files/ dir relative to role and/or play
       - /path/to/biz.txt
 
-- name: include tasks only if files exist.
-  include_tasks:
-    file: "{{ query('first_found', params) }}"
+- name: Set _found_file to the first existing file, or an empty list if no files found
+  set_fact:
+    _found_file: "{{ lookup('first_found', params) }}"
   vars:
     params:
       files:
-        - path/tasks.yaml
-        - path/other_tasks.yaml
+        - /path/to/foo.txt
+        - /path/to/bar.txt
+      paths:
+        - /extra/path/to/search
+      skip: True
+
+- name: Include tasks only if one of the files exist, otherwise skip the task
+  include_tasks:
+    file: "{{ item }}"
+  with_first_found:
+    files:
+     - path/tasks.yaml
+     - path/other_tasks.yaml
+    skip: True
+
+- name: Include tasks only if one of the files exists, otherwise skip
+  include_tasks: '{{ tasks_file }}'
+  when: tasks_file != ""
+  vars:
+    tasks_file: "{{ lookup('first_found', dict(files=['tasks.yaml', 'other_tasks.yaml']), errors='ignore') }}"
```

Comment:
again, this syntax we should phase out
```
asks_file: "{{ lookup('first_found', files=['tasks.yaml', 'other_tasks.yaml'], errors='ignore') }}"
```

Reply:
I've done this, but does this mean that "errors", "wantlist", "allow_unsafe", etc. are now reserved words; i.e. a plugin can't also use them to mean something different?  that didn't seem to be a problem when the arguments were passed as a separate dict.

Reply:
they always were, most lookups don't take a dict as argument, older lookups, like this one, had created their own interfaces, most others use a 'single string' k=v space separated options (also something we are moving away from)